py/server/deephaven/filters.py

```diff
@@ -80,21 +88,33 @@ def and_(filters: List[Filter]) -> Filter:
         filters (List[filter]): the component filters
 
     Returns:
-        a new Filter
+        a new and Filter
     """
     return Filter(j_filter=_JFilterAnd.of(*[f.j_filter for f in filters]))
 
 
 def not_(filter_: Filter) -> Filter:
-    """Creates a new filter that evaluates to true when the given filter evaluates to false.
+    """Creates a new filter that evaluates to the opposite of what filter_ evaluates to.
 
     Args:
         filter_ (Filter): the filter to negate with
 
     Returns:
-        a new Filter
+        a new not Filter
+    """
+    return filter_.not_()
+
+
+def is_null(col: str) -> Filter:
+    """Creates a new filter that evaluates to true when the col is null, and evaluates to false when col is not null.
+
+    Args:
+        col (str): the column name
+
+    Returns:
+        a new is-null Filter
     """
```

Comment:
Would ```Filter.from_("X == null")``` be equivalent to 'is_null("X")?

Reply:
It's logically the same, and will produce the same results, but the engine will do different things.

"X == null" will cause java to compile a filter kernel (ConditionFilter), whereas `is_null("X")` is able to construct a much more efficient `MatchFilter`.

Reply:
The more idiomatic way to express this via string would be `from_("isNull(X)")`, but that still results in ConditionFilter.

Reply:
How about adding an option in ```pattern``` to express the ``match null``` intention?

Reply:
"MatchFilter" is not a regex construction, it's a "match from this list of values" construction.